pkg/mcp/server/monitoring.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opencensus.io/stats/view"
+)
+
+const (
+	typeURL   = "typeURL"
+	errorCode = "code"
+	errorStr  = "error"
+	connectionID = "connectionID"
+)
+
+var (
+	TypeURLTag      tag.Key
+	ErrorCodeTag    tag.Key
+	ErrorTag        tag.Key
+	ConnectionIDTag tag.Key
+
+	// buckets are powers of 4
+	byteBuckets = []float64{1, 4, 16, 64, 256, 1024, 4096, 16384, 65536, 262144, 1048576, 4194304, 16777216, 67108864, 268435456, 1073741824}
+
+	ClientCount = stats.Int64(
+		"mcp/server/client_count",
```

Comment:
nit: I've been using the [Prometheus guidelines](https://prometheus.io/docs/practices/naming/#metric-names) on measure names.

So, instead of `client_count`, this would be something like: `clients_total`. And `request_acks` becomes something like `request_acks_total`, etc.

Note: This could be accomplished slightly differently if we want to keep the names of the `measures` here to be simple and update `newView` to pull from a map to prometheus appropriate names, a la: `viewName["client_count"] = "mcp/server/clients_total"`.

Reply:
I've mostly changed the names but am not sure if "message_size_bytes" is correct for a distribution.